test/completion.js

```diff
@@ -0,0 +1,148 @@
+      cmd += 'node ' + yocomplete + ' completion -- ' + yo + ' $cmd';
+
+      execFile('bash', ['-c', cmd], function (err, out) {
+        if (err) {
+          return done(err);
+        }
+
+        assert.ok(/-f/.test(out));
+        assert.ok(/--force/.test(out));
+        assert.ok(/--version/.test(out));
+        assert.ok(/--no-color/.test(out));
+        assert.ok(/--no-insight/.test(out));
+        assert.ok(/--insight/.test(out));
+        assert.ok(/--generators/.test(out));
+
+        done();
+      });
+    });
+  });
+
+  describe('Completer', function () {
+    beforeEach(function () {
+      this.completer = new Completer(this.env);
+    });
+
+    describe('#parseHelp', function () {
+      it('Returns completion items based on help output', function () {
+        var results = this.completer.parseHelp('backbone:app', help);
+        var first = results[0];
+
+        assert.equal(results.length, 6);
+        assert.deepEqual(first, {
+          name: '--skip-cache',
+          description: 'Do not remember prompt answers                         Default-> false'
+        });
+      });
+    });
+
+    describe('#item', function () {
+      it('Format results into { name, description }', function () {
+        var list = ['foo', 'bar'];
+        var results = list.map(this.completer.item('yo!', '--'));
+        assert.deepEqual(results, [{
+          name: '--foo',
+          description: 'yo!'
+        }, {
+          name: '--bar',
+          description: 'yo!'
+        }]);
+      });
+
+      it('Escapes certain characters before consumption by shell scripts', function () {
+        var list = ['foo'];
+
+        var desc = '#  yo I\'m a very subtle description, with chars that likely will break your Shell: yeah I\'m mean';
+        var expected = 'yo I m a very subtle description, with chars that likely will break your Shell-> yeah I m mean';
+        var results = list.map(this.completer.item(desc, '-'));
+
+        assert.equal(results[0].description, expected);
+      });
+    });
+
+    describe('#generator', function () {
+      it('Returns completion candidates from generator help output', function (done) {
+        // Here we test against yo --help (could use dummy:yo --help)
+        this.completer.generator({ last: '' }, function (err, results) {
+          if (err) {
+            return done(err);
+          }
+
+          /* eslint no-multi-spaces: 0 */
+          assert.deepEqual(results, [
+            { name: '--force',    description: 'Overwrite files that already exist' },
+            { name: '--version',  description: 'Print version' },
+            { name: '--no-color', description: 'Disable colors' },
+            { name: '-f',         description: 'Overwrite files that already exist' }
+          ]);
+
+          done();
+        });
+      });
+    });
+
+    describe('#complete', function () {
+      before(function (done) {
+        execFile('npm', ['install', 'generator-dummy', '-g'], done);
```

Comment:
Any chance that instead of running an npm install we could mock some functions? Maybe mock the return value of `getGeneratorsMeta`?

That'd make the test faster and less likely the flake.